test/Ocelot.UnitTests/Configuration/UpstreamHeaderRoutingOptionsCreatorTests.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using System.Linq;
+using Ocelot.Configuration.File;
+using Ocelot.Configuration.Creator;
+using Ocelot.Configuration;
+using Xunit;
+using TestStack.BDDfy;
+using Shouldly;
+
+namespace Ocelot.UnitTests.Configuration;
+
+public class UpstreamHeaderRoutingOptionsCreatorTests
+{
+    private FileUpstreamHeaderRoutingOptions _fileUpstreamHeaderRoutingOptions;
+    private IUpstreamHeaderRoutingOptionsCreator _creator;
+    private UpstreamHeaderRoutingOptions _upstreamHeaderRoutingOptions;
```

Comment:
I don't think I follow here. What do you suggest instead? Having the private helpers have an `out` argument? Or just flat out return the created object and the lambda in the call to `Given()` would assign the local variable? Just trying the first approach now and I'm getting errors about uninitialized variable. Since the compiler cannot determine the output of the lambdas, I'm going to have to assign _something_ to them. I suppose I could assign `null` but then I will be getting warnings about potential null references being used, I figure.

Reply:
Just fix merge conflicts plz, and we continue discussion...